### PR TITLE
Enable llama attention softmax in bf16

### PR DESCRIPTION
**Pull request by schoi-habana**
This PR adds an input arg to enable BF16 attention softmax layer for Llama finetuning.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by mandy-li — APPROVED** [review]
LGTM

**Review by regisss — COMMENTED** [review]
LGTM!
I would also add `--attn_softmax_bf16` to the extra arguments in the CI LoRA runs here: https://github.com/huggingface/optimum-habana/blob/main/tests/baselines/llama_7b.json

edit: done
